packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -147,6 +149,53 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
         e.preventDefault();
         const newNavigatedDate = navigateToNewDate(gridNavigatedDate, 'Day', -1, restrictedDatesOptions);
 
+        if (!!newNavigatedDate) {
+          setShouldFocusInDayGrid(true);
+          setGridNavigatedDate(newNavigatedDate);
+        }
+      },
+      moveToStartOfWeek: e => {
+        e.preventDefault();
+        const targetDate = getStartDateOfWeek(gridNavigatedDate, firstDayOfWeek);
+        const newNavigatedDate = contstraintNavigatedDate(gridNavigatedDate, targetDate, -1, restrictedDatesOptions);
+
+        if (!!newNavigatedDate) {
+          setShouldFocusInDayGrid(true);
+          setGridNavigatedDate(newNavigatedDate);
+        }
+      },
+      moveToEndOfWeek: e => {
+        e.preventDefault();
+        const targetDate = getEndDateOfWeek(gridNavigatedDate, firstDayOfWeek);
```

Comment:
Does it make sense to add the end/start of week navigation to the `NavigationKind` and then use the `navigateToNewDate` directly?

Reply:
Yeah, but I'll still need to export `contstraintNavigatedDate` for column navigation, since it would be based on the visible grid, and I don't want to overcomplicate `navigateToNewDate` with additional `firstDayOfWeek` param, so I'd rather keep it like that, WDYT?